docs/content/en/docs/pipeline-stages/builders/docker.md

```diff
@@ -67,15 +67,37 @@ These credentials are configured in the `cluster` section with the following opt
 
 {{< schema root="ClusterDetails" >}}
 
-To set up the credentials for Kaniko refer to the [kaniko docs](https://github.com/GoogleContainerTools/kaniko#kubernetes-secret) (**Note**: Rename the downloaded JSON key to *kaniko-secret* without appending *.json*).
-Alternatively, the path to a credentials file can be set with the `pullSecret` option:
+### Configure Kaniko Credentials 
+
+To set up the credentials for Kaniko refer to the [kaniko docs](https://github.com/GoogleContainerTools/kaniko#kubernetes-secret)
+
+(**Note**: Rename the downloaded JSON key to *kaniko-secret* without appending *.json*).
+
+Alternatively, the path to credentials file can be set with the `pullSecretPath` option:
 ```yaml
 build:
   cluster:
     pullSecretName: pull-secret-in-kubernetes
-    # OR
-    pullSecret: path-to-service-account-key-file
+    pullSecretPath: path-to-service-account-key-file-within-secret
+  
+```
+
+Skaffold can also set up credentials if a secret does not exist in your cluster. This is usually the case when you are performing a 
+build in your personal cluster. 
+
+First, [create a service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts#creating) 
+in the Google Cloud Console project you want to push the final image to with Storage Admin permissions. 
+Download a JSON key for this service account at a convenient location.
+```yaml
+build:
+  cluster:
+    pullSecretPath: path-to-local-service-account-key-file
```

Comment:
It's unfortunate this is called `pull`?

Reply:
This secret is only used for pulling images and we construct `GoogleApplicationCredentials` from it. 
Hence it was named as pull secret originally. But yes something we could re-think